### PR TITLE
Travel agent class

### PR DESCRIPTION
**Pull request by megan-venetianer**
# Description

This PR builds out the following methods on the TravelAgent class:
-findUser
-findTripRequests
-getYearlyIncome
-getTodaysTravelers

It also tests all of these methods and properties in the TravelAgent test file

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix or refactor (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Code for review (non-functional code that requires a code review)

## Checklist:

- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code if necessary

### CODE REVIEW

File: src/Travel-agent.js [hunk 1]
```diff
@@ -1,28 +1,61 @@
...
 
-  // should return user object by passing in the user's name. this function will return the id of the user and set it to this.id
-  findUser() {
-
+  findUser(travelerName, travelerData) {
+    let travelerInfo = travelerData.filter(traveler => {
+      return travelerName === traveler.name;
+    })
+    this.id = travelerInfo[0].id;
+    return travelerInfo[0].id
```
Comment (ERaith) [inline src/Travel-agent.js]: Remember to use a linter to check for missing ';'

File: src/Travel-agent.js [hunk 2]
```diff
@@ -1,28 +1,61 @@
...
 
   // total income generated this year (10% of user trip cost)
-  getTotalIncome() {
-
+  getYearlyIncome(tripsData, destinationData) {
+    let yearsTrips = [];
+    tripsData.forEach(trip => {
+      let thisYearsTrips = trip.date.split('');
+      if (thisYearsTrips[3] === '0' && trip.status === 'approved') {
+        yearsTrips.push(trip)
```
Comment (ERaith) [inline src/Travel-agent.js]: ;

File: src/Travel-agent.js [hunk 3]
```diff
@@ -1,28 +1,61 @@
...
 class TravelAgent extends User {
   constructor() {
     super();
     this.id;
   }
 
-  // should return user object by passing in the user's name. this function will return the id of the user and set it to this.id
-  findUser() {
-
+  findUser(travelerName, travelerData) {
```
Comment (ERaith) [inline src/Travel-agent.js]: Find user seems to return the id. 
Consider renaming the method to be: findUserID (Unless this is a placeholder)

File: test/TravelAgent-test.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+  it('should be an instance of travelAgent', function() {
+    expect(travelAgent).to.be.an.instanceof(TravelAgent);
+  });
+
+  it('should be able to find a traveler\'s id by their name', function() {
+    expect(travelAgent.findUser('Rachael Vaughten', travelerData)).to.equal(2)
+  })
+
+  it('should have an id property of the traveler they are searching for', function() {
```
Comment (ERaith) [inline test/TravelAgent-test.js]: Nice work testing these, when moving to fetch data from the remote server. You will want to test the sad paths of these if a user doesn't exist.

File: src/Travel-agent.js [hunk 5]
```diff
@@ -1,28 +1,61 @@
 import User from './User';
+var moment = require('moment');
 
 class TravelAgent extends User {
   constructor() {
     super();
```
Comment (ERaith) [inline src/Travel-agent.js]: user, travel-agent,traveler all have this.id.  You might want to add these to your super() and have these live on the user class.

File: src/Travel-agent.js (hunk 3 above)
Comment (megan-venetianer) [reply src/Travel-agent.js]: Yes, good point! I definitely meant to have it return the traveler object and forgot. I'll change that. Thanks!

File: test/TravelAgent-test.js (hunk 4 above)
Comment (megan-venetianer) [reply test/TravelAgent-test.js]: Agreed, thanks for the reminder!

File: src/Travel-agent.js (hunk 5 above)
Comment (megan-venetianer) [reply src/Travel-agent.js]: Great point. I think that was my initial intention, and somehow missed actually adding it to the super somewhere along the way.

**Pull request merged**